kernel/src/hil/sensors.rs

```diff
@@ -31,6 +31,18 @@ pub trait HumidityClient {
     fn callback(&self, value: usize);
 }
 
+/// A basic interface for a proximity sensor
+pub trait ProximityDriver<'a> {
+    fn set_client(&self, client: &'a dyn ProximityClient);
+    fn read_proximity(&self) -> ReturnCode; // Instantaneous proximity reading
+    fn read_proximity_on_interrupt(&self, low: u8, high: u8) -> ReturnCode; // Proximity reading returned only after interrupt is detected
+}
+
+pub trait ProximityClient {
+    /// Called when proximity reading has completed (`command_type` is used by proximity.rs to match callbacks to the apps issuing the appropriate commands)
+    fn callback(&self, value: usize, command_type: usize);
```

Comment:
What are the units of `value`? What are the possible values of `command_type`? Ideally, the HIL should not be specific to any particular capsule or underlying driver.

Reply:
@bradjc and I have suggested "value" should be unitless as it is impractical to use a formula of some sort to convert the proximity data to inches or millimeters because this reading depends on a multitude of factors (size of object, whether it emits infrared radiation, distance).  Looking around, most proximity sensing IC's use 8 bit registers to store their proximity readings so the range of a possible reading would be in 8 bit resolution (0 - 255) with 255 being the closest measurable distance.  If other drivers and IC's posses different resolutions, I suppose they can do some shifting to get the data into 0 - 255 range.  Ex: if 10-bit resolution then data = data>>2 so 1023 becomes 255.

Reply:
That seems reasonable to me, thanks for explaining.